src/test/java/com/bankofspring/hello/HelloTest.java

```diff
@@ -0,0 +1,48 @@
+package com.bankofspring.hello;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+
+/**
+ * @author slc
+ *
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration("classpath:hello/hello-applicationContext.xml")
+public class HelloTest {
+
+	@Autowired
+	private String article;
+	
+	@Autowired
+	@Qualifier("annotationSimpleHelloImpl")
+	private HelloService annotationHelloService;
+	
+	@Autowired
+	@Qualifier("xmlSimpleHelloImpl")
+	private HelloService xmlHelloService;
+	
+	@Test
+	public void testAnnotationHelloService() {
+		assertThat( annotationHelloService.sayHello(), containsString( article ) );
```

Comment:
I really like the way you've done this, ensuring that you autowire article. 

What message do you get if the assert fails compared with if you used the assertThat( "failure message", annotationHelloService.sayHello(), containsString( article ) );?